src/book/03-guides/01-main-codebase/02-maintaining-subsystems.mdx

```diff
@@ -48,12 +49,50 @@ The resulting network should be exported to a [yaml](https://yaml.org/) file and
 
 3. Add this configuration file to the NUbots repository in the [VisualMesh module](https://github.com/NUbots/NUbots/tree/main/module/vision/VisualMesh/data/config/networks). Replace or add a configuration file depending on the use case of the Mesh - `RobocupNetwork.yaml` is for soccer playing on the real robot and `WebotsNetwork` is for soccer playing in the Webots simulator. View the [Git Guide](/guides/general/git) for information on using Git and submitting this change in a pull request.
 
+#### Running OpenCL not on the robot
+
+The Visual Mesh works best running the OpenCL framework, however most operating systems are not set up to run this by default. Whilst running the Visual Mesh on CPU is possible, sometimes it is best to replicate an environment as close to what is running on the robot as possible. Follow the steps below to set up OpenCL on a device other than a robot.
+
+1. Install the OpenCL SDK for your device from the [Intel website](https://www.intel.com/content/www/us/en/developer/tools/opencl-sdk/choose-download.html)
+
+2. Create a temporary directory and install the pre-built OpenCL drivers
+
+```bash
+mkdir temp
+cd temp
+wget https://github.com/intel/compute-runtime/releases/download/19.14.12751/intel-gmmlib_19.1.1_amd64.deb
+wget https://github.com/intel/compute-runtime/releases/download/19.14.12751/intel-igc-core_19.11.1622_amd64.deb
+wget https://github.com/intel/compute-runtime/releases/download/19.14.12751/intel-igc-opencl_19.11.1622_amd64.deb
+wget https://github.com/intel/compute-runtime/releases/download/19.14.12751/intel-opencl_19.14.12751_amd64.deb
+wget https://github.com/intel/compute-runtime/releases/download/19.14.12751/intel-ocloc_19.14.12751_amd64.deb
+sudo apt install ./*deb
+```
+
+3. Change the `VisualMesh.yaml` configuration "engine" node to `opencl`
+
+4. Build the binary and then run with the following command to bind the PC drivers to the docker container
+
+```bash
+./b build
+./b run --mount type=bind,source="/dev/dri",target="/dev/dri" <role_name>
+```
+
 ### Camera Calibration
 
 The vision system cannot work optimally if the cameras are not calibrated correctly. [The input page](/system/subsystems/input) describes the camera parameters that can be calibrated.
 
 An automatic camera calibration tool is available in the NUbots repository. See the [camera calibration guide](/guides/main/camera-calibration) to find out how to use this tool.
 
+### Fake Camera
+
+To run the fake camera module, an images directory needs to exist inside the `data` directory of the module.
+
+1. Create an `images` directory and add several images with filenames matching `imageXXXXXX.jpg` where "XXXXXX" refers to any numerical digits
```

Comment:
Does it have to be 6 digits? If so this would help clarify:

```suggestion
1. Create an `images` directory and add several images with filenames matching `imageXXXXXX.jpg`, where "XXXXXX" is any 6-digit number, e.g. `000001`,  `000002`, etc.
```

Reply:
Doesn't have to be 6 digits, I believe it just needs to be enumerated somehow